lib/console.js

```diff
@@ -249,6 +249,75 @@ Console.prototype.groupEnd = function groupEnd() {
     this[kGroupIndent].slice(0, this[kGroupIndent].length - 2);
 };
 
+const untablableTypes = [
+  'symbol', 'undefined', 'boolean', 'number', 'string', 'function',
+];
+
+const untablableValues = [
+  undefined, null, NaN, true, false, global, process,
+];
+
+const untablable = (value) =>
+  untablableValues.includes(value) ||
+  untablableTypes.includes(typeof value);
+
+const { isSet, isMap, isSetIterator, isMapIterator } = process.binding('util');
+function getTablableKeysOf(O) {
+  if (isSet(O))
+    return Array.from({ length: O.size }, (e, i) => i);
+  if (isSetIterator(O))
+    return Array.from(O).map((i) => i - 1);
```

Comment:
This will exhaust `O`, preventing it from further usage; use the `previewMapIterator` and `previewSetIterator` from `internal/v8` instead. Make sure to set a limit in how many elements to allow. See how it's used in `util.format` (the limit is 100 in this case):

https://github.com/nodejs/node/blob/c84582cbb6362fa8b2eb2d3bc153617fef2982d1/lib/util.js#L821-L840